src/org/zaproxy/zap/extension/ascanrulesBeta/Csrftokenscan.java

```diff
@@ -135,7 +138,11 @@ public String getReference() {
 	 */
 	@Override
 	public void init() {
-
+		String ignoreConf = Model.getSingleton().getOptionsParam().getConfig().
+				getString("rules.csrf.ignorelist");
+		if (ignoreConf != null && ignoreConf.length() > 0) {
+			ignoreList = ignoreConf.split(",");

```

Comment:
It could also clean the values (e.g. trim, remove empty values), it would not longer need to do that in `formOnIgnoreList`.